_posts/2019-07-08-persistent-storage-improvements.md

```diff
@@ -0,0 +1,261 @@
+This blog post will tell you more about the improvements to persistent storage in the 0.12 release.
+
+<!--more-->
+
+##  Resizing persistent volumes
+
+Support for resizing volumes depends on the version of your Kubernetes or OpenShift cluster and on the infrastructure it runs on.
+We tested this feature with Kubernetes and OpenShift on Amazon AWS.
+It should be compatible with Kubernetes 1.11+ and OpenShift 3.11+.
+Resizing persistent volumes should work on most major public clouds (such as Amazon AWS, Microsoft Azure and Google Cloud) and many other storage types (such as Cinder or Ceph).
+
+To tell your Kubernetes or OpenShift cluster that your storage supports volume resizing, you have to set the `allowVolumeExpansion` option in your StorageClass to `true`.
+For example following StorageClass would create Amazon AWS GP2 volumes with `xfs` filesystem, encryption and enabled volume expansion:
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: ssd
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
+provisioner: kubernetes.io/aws-ebs
+parameters:
+  type: gp2
+  encrypted: "true"
+  fsType: "xfs"
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+The option is called `allowVolumeExpansion` because right now you can only increase the size of the volumes to expand them - you cannot decrease it.
+Strimzi will check that the storage class has this option set to `true` before it tries to resize any volumes.
+The [original blog post](https://strimzi.io/2019/02/28/resizing-persistent-volumes.html) explained in detail how does the resizing work:
+
+1. The size increase is requested by changing the `spec.resources.requests.storage` of the Persistent Volume Claim (PVC)
+2. Kubernetes will request resizing of the volumes from your infrastructure
+3. Once the resizing of the volume is finished, the pod using this volume needs to be restarted to allow the expansion of the file system
+
+All what Strimzi does is that it simplifies this process and takes care of it for you.
+As a user all you need to do is edit your Kafka custom resource and increase the requested storage size.
+You can increase the size of both Zookeeper and Kafka volumes.
+For example you can change the resource from:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 100Gi
+      deleteClaim: false
+  zookeeper:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 100Gi
+      deleteClaim: false
+```
+
+to:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 200Gi
+      deleteClaim: false
+  zookeeper:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 1000Gi
```

Comment:
more realistically do we want to do the opposite increase? 1000 Gi for Kafka and 200 Gi for ZooKeeper, just because in Kafka you store messages so would make sense having bigger size there.